libraries/AP_Airspeed/AP_Airspeed_NMEA.cpp

```diff
@@ -42,6 +42,8 @@ bool AP_Airspeed_NMEA::init()
         return false;
     }
 
+    set_bus_id(AP_HAL::Device::make_bus_id(AP_HAL::Device::BUS_TYPE_SERIAL,0,0,0));
```

Comment:
This `set_bus_id` should probably be done in the base class.  So `uint32_t make_bus_id() = 0` is implemented by all backend subclasses and `AP_Airspeed_Backend`'s `init` does `set_bus_id(make_bus_id())`

This ensures the bus ID will always be set.

... on further reading, the UAVCAN backend screws this idea up as the driver doesn't have the required information to set the bus id.